### PR TITLE
feat(cluster): add the option for a custom node selector in scaleReads

### PR DESCRIPTION
**Pull request by ramonsnir**
#250

### CODE REVIEW

**Comment by luin** [comment]
Awesome! Thank you for the pull request!

It seems better to change `function(slaves, command) {}` to `function(nodes, command) {}` where the first  element of `nodes` is the master node and the reset are slaves so that it's possible to send the queries to masters.

File: lib/cluster/index.js [hunk 1]
```diff
@@ -453,15 +454,28 @@ Cluster.prototype.sendCommand = function (command, stream, node) {
...
-              key = utils.sample(nodeKeys, 1);
+            if (typeof to === 'function') {
+              var nodes =
+                  nodeKeys
+                  .slice(1)
+                  .map(function(key) {
+                    return _this.connectionPool.nodes.all[key];
+                  });
+              redis = to(nodes, command);
+              if (redis.constructor === Array) {
```
Comment (luin) [inline lib/cluster/index.js]: `if (Array.isArray(redis))` may be better IMO, and we should check whether `redis` is null/undefined and default to the master node.

Comment (ramonsnir) [inline lib/cluster/index.js]: Yes, you are correct. I'm sadly used too much to client-side JS, and Array.isArray is unavailable in some MSIE versions...

**Comment by luin** [comment]
LGTM. Any thoughts about changing `function(slaves, command) {}` to `function(nodes, command) {}` so that we don't need to call `slice(1)` everytime.

**Comment by ramonsnir** [comment]
It bothered me too... I'll do that, and make sure the README states it clearly.

**Comment by luin** [comment]
Cool! Merged. :+1:

**Comment by ramonsnir** [comment]
Thanks!

File: lib/cluster/index.js (hunk 1 above)
Comment (BridgeAR) [inline lib/cluster/index.js]: As a hint: there's a easy [polyfill](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#Polyfill) for this and Microsoft itself does [not maintain IE 8](https://support.microsoft.com/en-us/lifecycle#gp/Microsoft-Internet-Explorer) or older, not even with security fixes on any regular Windows version. Most libraries drop support for it or have already dropped the support and the faster all developers switch to not supporting it anymore, the faster no one is going to visit any website with it anymore.

**Pull request merged**